.github/workflows/build-rc-app.yml

```diff
@@ -13,7 +13,7 @@ on:
     types: [ opened, synchronize ]
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.event.merge_group.head_sha }}
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
This will be different for each case, and the same for each branch. 
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

The fully-formed ref of the branch or tag that triggered the workflow run: 
- For workflows triggered by push, this is the branch or tag ref that was pushed
- For workflows triggered by pull_request, this is the pull request merge branch.
- For merge_groups, something like, `gh-readonly-queue/release/candidate/pr-3209-3321e269be34f4ec128ce167ae67c7375cf237f3`

Reply:
I don't think it helps directly to the merge group issue we are having, but it makes things more straightforward with one var.